### PR TITLE
Add zerank selected-logit reranking adapter

### PR DESCRIPTION
**Pull request by nisavid**
## Summary

Adds a Lemonade-side reranking adapter for ZeroEntropy's `zerank-2` GGUF model now that the installed `llama.cpp` packages expose selected-token logits on `/completion`.

The adapter builds the ZeroEntropy prompt, requests a single-token completion with `token_logits` for the configured true-token id, ranks documents by the returned raw logit, and returns OpenAI-compatible reranking results with normalized relevance scores.

## Changes

- Add `zeroentropy-logit-score` llama.cpp reranking adapter support.
- Register built-in `zerank-2-GGUF` with `mradermacher/zerank-2-GGUF:Q8_0`.
- Auto-detect imported `zerank-2` llama.cpp models as reranking models and preserve the needed recipe options.
- Normalize stale saved `zerank-2` labels so existing user entries route as reranking rather than chat models.
- Add focused C++ adapter coverage and an opt-in live server test for the installed-package path.

## Verification

- `cmake --build --preset default --target lemond`
- `g++ -std=c++17 -I src/cpp/include test/cpp/test_llamacpp_reranking_adapter.cpp src/cpp/server/backends/llamacpp_reranking_adapter.cpp src/cpp/server/recipe_options.cpp -o /tmp/test_llamacpp_reranking_adapter && /tmp/test_llamacpp_reranking_adapter`
- `python -X pycache_prefix=/tmp/lemonade-pycache -m py_compile test/server_llm.py`
- `python -m json.tool src/cpp/resources/server_models.json`
- `git diff --check`
- Installed-package live smoke using `llama.cpp-vulkan-gfx1151 b9222-3`:
  - server: `build/lemond /tmp/lemonade-zerank-installed-vulkan --port 18082 --host 127.0.0.1`
  - request: `/api/v1/reranking` with query `capital of France`
  - result ranked the Paris document first: `index: 1`, score about `0.96765`
- Opt-in live test passed against the same server:
  - `LEMONADE_TEST_ZERANK_ADAPTER=1 LEMONADE_TEST_ZERANK_MODEL=zerank-2-GGUF`

## Notes

Depends on the merged/installed `llama.cpp` selected-logits package work:

- `llama.cpp-vulkan-gfx1151 b9222-3`
- `llama.cpp-hip-gfx1151 b9222-3`

## Summary by CodeRabbit

* **New Features**
  * Added ZeroEntropy logit-score reranking adapter for document relevance ranking and UI settings to configure adapter, token ID, and logit scale
  * Registered a zerank-2 reranking model variant with default recipe options and automatic detection in model discovery/registration
  * Reranking endpoint updated to route adapter-mode requests while preserving native reranking behavior

* **Tests**
  * Added unit and end-to-end tests covering adapter logic, scoring, sorting, and error handling

[Review Change Stack](https://app.coderabbit.ai/change-stack/nisavid/lemonade/pull/17?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by nisavid** [comment]
@coderabbitai review
